src/NuGet.Core/NuGet.DependencyResolver.Core/GraphModel/GraphOperations.cs

```diff
@@ -774,6 +805,93 @@ private static CyclesAndDowngrades CreateState(List<GraphNode<RemoteResolveResul
             };
         }
 
+        private static bool TryResolveConflictsForCPVM<TItem>(this GraphNode<TItem> root, List<VersionConflictResult<TItem>> versionConflicts)
```

Comment:
Doing a diff between `TryResolveConflictsForCPVM` and `TryResolveConflicts`, there are only 2 differences:

1. `ForCPVM` version calls an overload of `ForEach` with an extra skipNode parameter. However, the `ForEach` overload with `skipNode` calls the overload with `skipNode` with a `null` value. Therefore, if this method had the `skipNode` func as a method parameter, then both the CPVM and non-CPVM codepaths could use the same codepath.

2. CPVM version calls `RejectCentralTransitiveBecauseOfRejectedParents`. `RejectCentralTransitiveBecauseOfRejectedParents` looks like a no-op for non-CPVM graphs. Alternatively we chould guard the method call in an `if` statement.

It looks to me like we can get rid of this new method, which I think is important because it implements business logic, (not just trivial code), so this type of code duplication is real technical debt that can lead to bugs if one of the two implementations change in the future.

Reply:
I would let the two methods different. In my opinion it is easier to follow the logic in this way.

Reply:
Updated.